akka-actor-typed/src/main/scala/akka/actor/typed/pubsub/Topic.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.pubsub
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.actor.typed.internal.pubsub.TopicImpl
+import akka.actor.typed.scaladsl.Behaviors
+import akka.annotation.DoNotInherit
+
+import scala.reflect.ClassTag
+
+/**
+ * A pub sub topic is an actor that handles subscribing to a topic and publishing messages to all subscribed actors.
+ *
+ * It is mostly useful in a clustered setting, where it is intended to be started once on every node that want to
+ * house subscribers or publish messages to the topic, but it also works in a local setting without cluster.
+ *
+ * In a clustered context messages are deduplicated so that there is at most one message sent to each node for
+ * each publish and if there are no subscribers on a node, no message is sent to it. Note that the list of subscribers
+ * is eventually consistent and there are no delivery guarantees built in.
+ *
+ * Each topic results in a [[akka.actor.typed.receptionist.ServiceKey]] in the [[akka.actor.typed.receptionist.Receptionist]]
+ * so the same scaling recommendation holds for topics, see docs:
+ * https://doc.akka.io/docs/akka/current/typed/actor-discovery.html#receptionist-scalability
+ */
+object Topic {
+
+  /**
+   * Not for user extension
+   */
+  @DoNotInherit
+  sealed trait Command[T] extends TopicImpl.Command[T]
+
+  /**
+   * Publish the message to all currently known subscribers.
+   */
+  final case class Publish[T](message: T) extends Command[T]
+
+  /**
+   * Send the message to one randomly choosen subscriber.
+   */
+  final case class Send[T](message: T, preferLocal: Boolean) extends Command[T]
+
+  /**
+   * Subscribe to this topic. Should only be used for local subscribers.
+   */
+  final case class Subscribe[T](subscriber: ActorRef[T]) extends Command[T]
+
+  /**
+   * Unsubscribe a previously subscribed actor from this topic.
+   */
+  final case class Unsubscribe[T](subscriber: ActorRef[T]) extends Command[T]
+
+  /**
+   * Scala API: Create a topic actor behavior for the given topic name and message type.
+   */
+  def apply[T <: AnyRef](topicName: String)(implicit classTag: ClassTag[T]): Behavior[Command[T]] =
```

Comment:
can't the ClassTag be in `[T : ClassTag]`

why AnyRef?

Reply:
I just like it better as an obvious implicit than a magic one, can change if important.

Got some type bound issues at some point, with `T` inferred to `Nothing` but maybe that was caused by something else. I'll try removing it again.

Reply:
no strong opinion

Reply:
To the ack point, I think some of the acks in classic are determined by a `replyTo: Option[ActorRef]`. I would consider if the user might need, otherwise drop offering it.

Reply:
Probably best to make the message case classes non public API so we can add things after the fact.